src/os/bluestore/BlueStore.cc

```diff
@@ -4160,6 +4160,8 @@ int BlueStore::_write_bdev_label(CephContext *cct,
   if (r < 0) {
     derr << __func__ << " failed to write to " << path
 	 << ": " << cpp_strerror(r) << dendl;
+    VOID_TEMP_FAILURE_RETRY(::close(fd));
```

Comment:
Instead, can you 'goto out', and add an out: label on the VOID_TEMP_FAILURE... line below?